src/language-markdown/print-paragraph.js

```diff
@@ -0,0 +1,54 @@
+import { fill } from "../document/builders.js";
+import { DOC_TYPE_ARRAY, DOC_TYPE_FILL } from "../document/constants.js";
+import { getDocType } from "../document/utils.js";
+
+/**
+ * @typedef {import("../common/ast-path.js").default} AstPath
+ * @typedef {import("../document/builders.js").Doc} Doc
+ */
+
+/**
+ * @param {AstPath} path
+ * @param {*} options
+ * @param {*} print
+ * @returns {Doc}
+ */
+function printParagraph(path, options, print) {
+  const parts = path.map(print, "children");
+  return flattenFill(parts);
+}
+
+/**
+ * @param {Doc[]} docs
+ * @returns {Doc}
+ */
+function flattenFill(docs) {
+  /** @type {Doc[][]} */
+  const parts = [[]];
+
+  /**
+   * @param {Doc[]} docArray
+   */
+  (function rec(docArray) {
+    for (const doc of docArray) {
+      const docType = getDocType(doc);
+      if (docType === DOC_TYPE_ARRAY) {
+        rec(doc);
+        continue;
+      }
+
+      let head = doc;
+      let rest = [];
+      if (docType === DOC_TYPE_FILL) {
+        [head, ...rest] = doc.parts;
```

Comment:
I'm not sure about this part, do we need go deeper? `head` and docs in `rest` can also be `DOC_TYPE_ARRAY` or `DOC_TYPE_FILL`.

Reply:
I'm going to run some test first, https://github.com/prettier/prettier-regression-testing/issues/435#issuecomment-1908654047

Reply:
I'm not very sure it. But probably `fill()` isn't nested in paragraph. If we want to go deeper, we should take care that we should go deeper only even (odd for `doc.parts`) index element.